spec/integration/publishing_step_events_spec.rb

```diff
@@ -19,7 +19,27 @@
     end.new
   }
 
-  let(:subscriber) { spy(:subscriber) }
+  let(:subscriber) do
+    Class.new do
+      attr_reader :started, :success, :failed
+
+      def initialize
+        @started = []
+        @success = []
+        @failed = []
+      end
+
+      def on_step(event)
+        started << event[:step_name]
+      end
+      def on_step_succeeded(event)
+        success << {step_name: event[:step_name], args: event[:args]}
+      end
+      def on_step_failed(event)
+        failed << {step_name: event[:step_name], args: event[:args], value: event[:value]}
+      end
+    end.new
+  end
```

Comment:
What's your reasoning for building up a full subscriber class like this instead of just using the spy?